### PR TITLE
profiles: add prefix llvm profile for macos

### PR DESCRIPTION
**Pull request by DarthGandalf**
Bug: https://bugs.gentoo.org/758167

### CODE REVIEW

File: profiles/features/llvm/make.defaults [hunk 1]
```diff
@@ -14,7 +14,8 @@ LD="ld.lld"
 # We don't set LDFLAGS here for linker, rtlib, etc because clang-common enforces
 # the needed bits for us and it confuses clang->gcc for stuff like Fortran,
 # see bug #908798.
-LDFLAGS="${LDFLAGS} -Wl,--as-needed"
+# FIXME: no idea where this belongs
+#LDFLAGS="${LDFLAGS} -Wl,--as-needed"
```
Comment (thesamesam) [inline profiles/features/llvm/make.defaults:18]: I guess we can try filter -Wl,--as-needed out in LDFLAGS via profile.bashrc in the Darwin clang profiles, or we can just set LDFLAGS directly in make.defaults for now in our Darwin clang profiles (maybe let's do that for now?)

Comment (DarthGandalf) [reply profiles/features/llvm/make.defaults]: I got another idea how to fix this. In my preliminary testing it works.

File: profiles/prefix/darwin/macos/14.0/arm64/clang/make.defaults [hunk 2]
```diff
@@ -0,0 +1,4 @@
+LD=ld64.lld
```
Comment (grobian) [inline profiles/prefix/darwin/macos/14.0/arm64/clang/make.defaults:1]: it feels like most of this stuff belongs in some generic place, e.g. under profiles/prefix/darwin/macos/features/clang

then of course link to it, such that macos 15, 16, etc. all can just link to this set of changes instead redefining them again, including of course support for x64, etc.

Comment (DarthGandalf) [reply profiles/prefix/darwin/macos/14.0/arm64/clang/make.defaults]: I think done. I don't know whether ld64 is used on x64 though

**Comment by DarthGandalf** [comment]
After merging this please make a new snapshot, and I'll retest changes to bootstrap script, then send that PR.

**Comment by grobian** [comment]
I think clang-ld64 is a misnomer -> you're NOT using ld64, but lld.  A better name for the feature profile would thus probably be clang-lld.  clang-ld64 would be the combination of llvm + binutils-apple/darwin-xtools, which I somehow doubt will ever work.

**Comment by DarthGandalf** [comment]
Moved

File: profiles/prefix/darwin/macos/features/clang-lld/make.defaults [hunk 3]
```diff
@@ -0,0 +1,4 @@
+LD=ld64.lld
```
Comment (grobian) [inline profiles/prefix/darwin/macos/features/clang-lld/make.defaults:1]: can you add a copyright to this file?

File: profiles/prefix/darwin/macos/features/clang-lld/package.mask [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# If installed, causes some miscompilation somewhere, and cmake crashes during bootstrap
```
Comment (grobian) [inline profiles/prefix/darwin/macos/features/clang-lld/package.mask:1]: please add copyright to this file

File: profiles/prefix/darwin/macos/features/clang-lld/packages [hunk 5]
```diff
@@ -0,0 +1 @@
+-*sys-devel/binutils-apple
```
Comment (grobian) [inline profiles/prefix/darwin/macos/features/clang-lld/packages:1]: please add copyright to this file

**Comment by grobian** [comment]
please add the copyrights: ```
# Copyright 1999-2024 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2
```
Then everything is fine from my point of view to merge.

Thanks!

**Comment by DarthGandalf** [comment]
Added.

**Comment by thesamesam** [comment]
> After merging this please make a new snapshot, and I'll retest changes to bootstrap script, then send that PR.

Merged -- @grobian could you handle the new snapshot? thank you!

**Comment by grobian** [comment]
> > After merging this please make a new snapshot, and I'll retest changes to bootstrap script, then send that PR.
> 
> Merged -- @grobian could you handle the new snapshot? thank you!

FYI: Handling the snapshot nowadays is done by simply changing the date in the bootstrap-prefix.sh script.  It's retrieved from the prefix rsync mirrors, and the tar is kept for as long as it's referenced in the script.  So, no more uploading etc. is necessary.  Just point to an existing snapshot date.

**Comment by thesamesam** [comment]
Oh, thank you!

**Pull request closed without merging**